packages/core-transactions/src/handlers/transaction.ts

```diff
@@ -37,39 +37,37 @@ export abstract class TransactionHandler implements ITransactionHandler {
 
     public canBeApplied(
         transaction: Interfaces.ITransaction,
-        wallet: State.IWallet,
-        databaseWalletManager: State.IWalletManager,
-    ): boolean {
-        const { data }: Interfaces.ITransaction = transaction;
+        sender: State.IWallet,
+        databaseWalletManager?: State.IWalletManager,
+    ): void {
+        const data: Interfaces.ITransactionData = transaction.data;
```

Comment:
Keep the destructure.

Reply:
The reason for my change is to make it clear to the reader what is the type of the `data` variable.
The previous code `const { data }: Interfaces.ITransaction = transaction;` not only does not show the type of the `data` variable, but it misleadingly contains another type `Interfaces.ITransaction` which is NOT the type of `data`.

Reply:
How is it misleading unless you don't understand how destructuring works. It doesn't `contain another type` at all. This just duplicates the field name for no reason.

Reply:
`const { variable }: Type = expression;` is misleading because `Type` is not the type of `variable`.